client/asset/eth/eth.go

```diff
@@ -2402,18 +2520,377 @@ func (eth *ETHWallet) checkForNewBlocks(reportErr func(error)) {
 	eth.log.Debugf("tip change: %s (%s) => %s (%s)", prevTip.Number,
 		currentTipHash, bestHdr.Number, bestHash)
 
+	walletList := eth.walletList()
+
 	go func() {
-		for _, w := range eth.walletList() {
+		for _, w := range walletList {
 			w.tipChange(nil)
 		}
 	}()
 	go func() {
-		for _, w := range eth.walletList() {
+		for _, w := range walletList {
 			w.checkFindRedemptions()
 		}
 	}()
 }
 
+// getLatestMonitoredTx looks up a txHash in the monitoredTxs map. If the
+// transaction has been replaced, the latest in the chain of transactions
+// is returned.
+//
+// w.monitoredTxsMtx MUST be held when calling this function.
+func (w *assetWallet) getLatestMonitoredTx(txHash common.Hash) (*monitoredTx, error) {
+	tx, found := w.monitoredTxs[txHash]
+	maxLoops := 100 // avoid an infinite loop in case of a cycle
+	for i := 0; i < maxLoops; i++ {
+		if !found {
+			return nil, fmt.Errorf("%s not found among monitored transactions", txHash)
+		}
+		if tx.replacementTx == nil {
+			return tx, nil
+		}
+		tx, found = w.monitoredTxs[*tx.replacementTx]
+	}
```

Comment:
So we need to track the entire chain, or just the most recent?

Reply:
It's tracking the whole chain, but once every match using this tx has called the function, and the redemption coin IDs are updated in the matches, the old ones are not required anymore.